broker.go

```diff
@@ -464,13 +498,15 @@ func (b *Broker) responseReceiver() {
 		}
 
 		buf := make([]byte, decodedHeader.length-4)
-		_, err = io.ReadFull(b.conn, buf)
+		remainingPayloadSize, err := io.ReadFull(b.conn, buf)
```

Comment:
You don't need to capture this value (or `headerSize` above) since `ReadFull` will return an error unless it completely fills the buffer, and we'll never even get to the metric call in that case. You should be able to use `len(header) + len(buf)` instead.